### PR TITLE
✨ Make UI Flow More Intuitive

### LINKED ISSUE #79: UI flows not intuitive - sometimes the user is delegated to the "wrong" dialog
#### Describe your issue

- We need a concept what should happens next (BPMN-Diagrams)
- Test the current flow and fix it

#### What's your setup

- OS (`Windows/OS X/Linux` + `version`): macOS High Sierra
- Browser: Safari

#### Issue checklist

Please check the boxes in this list after submitting your Issue:

- [x] I've checked if this issue already exists
- [ ] I've included all the information that i think is relevant
- [ ] I've added logs and/or screenshots (if applicable)
- [x] I've mentioned PRs and issues that relate to this one

### PR DESCRIPTION
**Pull request by Vyperus**
## What did you change?

This Branch adds the "Waiting Room".

If the User proceeds a task, in any way, he won't be delegated to the Task List directly.
He will be delegated to the "Waiting Room" where Charon will listen to the "renderUserTask"-Event and "processEnd"-Event to redirect the user to the correct view.

If there is an "renderUserTask"-Event, while the user is waiting, he will be redirected to the new Task in the dynamic-ui.

If there is an "processEnd"-Event he will be redirected to the Task-List automatically.

Closes #79.
 
## How can others test the changes?

Switch to this branch and start Charon with `npm install` and `npm start`.
Now navigate to `localhost:9000`.
Create a new UserTask by starting the Create-Process. (Don't forget to put "persist" on "true")

You should be redirected to the form to enter the data and, after you clicked "Abbrechen", you will be directly (after a short stop at the waiting room) redirected to the next Task (confirming that the process wasn't created).

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.)
- [x] I've merged the `develop` branch into my branch before finishing this PR.
- [x] I've **not added any other changes** than the ones described above.
- [x] I've mentioned all **PRs, which relate to this one**

### CODE REVIEW

File: src/modules/waiting-room/waiting-room.html [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<template>
+  <require from="./waiting-room.css"></require>
+  <div class="waiting-room__grid">
+    <h2 class="waiting-room__title">Waiting for UserTasks</h2>
```
Comment (heikomat) [inline src/modules/waiting-room/waiting-room.html]: This is to technical. Maybe write something like `process is executing` or `continuing process`

File: src/modules/waiting-room/waiting-room.html [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<template>
+  <require from="./waiting-room.css"></require>
+  <div class="waiting-room__grid">
+    <h2 class="waiting-room__title">Process is executing</h2>
+    <img class="waiting-room__spinner" src="src/resources/images/rolling.svg">
+    <button class="btn btn-danger waiting-room__cancel-button" click.delegate="navigateToTaskList()" title="Dies unterbricht den Prozess nicht!">Nicht warten</button>
```
Comment (heikomat) [inline src/modules/waiting-room/waiting-room.html]: Charon ist englisch, nur die BPMN-Prozesse in Charon sind deutsch. Hier bitte englisch verwenden.

File: src/modules/waiting-room/waiting-room.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  private renderUserTaskCallback: any = (userTaskConfig: IUserTaskConfig): void => {
+    toastr.success('Prozess fortgesetzt');
+    if (userTaskConfig.userTaskEntity.process.id === this.processInstanceId) {
+      this.router.navigate(`/task/${userTaskConfig.id}/dynamic-ui`);
+      this.consumerClient.off('renderUserTask', this.renderUserTaskCallback);
+    }
+  }
+
+  private processEndCallback: any = (processInstanceId: string): void => {
+    toastr.warning('Prozess beendet');
```
Comment (heikomat) [inline src/modules/waiting-room/waiting-room.ts]: Englisch verwenden

File: src/modules/waiting-room/waiting-room.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  private consumerClient: ConsumerClient;
+  private processInstanceId: string;
+
+  constructor(router: Router, consumerClient: ConsumerClient) {
+    this.router = router;
+    this.consumerClient = consumerClient;
+  }
+
+  private renderUserTaskCallback: any = (userTaskConfig: IUserTaskConfig): void => {
+    toastr.success('Prozess fortgesetzt');
```
Comment (heikomat) [inline src/modules/waiting-room/waiting-room.ts]: Englisch verwenden

File: src/resources/images/rolling.svg [hunk 5]
```diff
@@ -0,0 +1,5 @@
+<svg width="200px" height="200px" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 100 100" preserveAspectRatio="xMidYMid" class="lds-rolling" style="animation-play-state: running; animation-delay: 0s; background: none;">
```
Comment (Paulomart) [inline src/resources/images/rolling.svg]: Could we not use font-awesome?

File: aurelia_project/aurelia.json [hunk 6]
```diff
@@ -230,6 +230,15 @@
...
             ]
+          },
+          {
+            "name": "toastr",
+            "path": "../node_modules/toastr",
+            "main": "toastr",
+            "resources": [
+              "build/toastr.min.css"
+            ],
+            "deps": ["jquery"]
```
Comment (cmg-dev) [inline aurelia_project/aurelia.json]: :+1:

**Comment by cmg-dev** [comment]
@moellenbeck @ppoetschke 

Is this okay from your perspective?

**Pull request merged**